astroquery/gaia/core.py

```diff
@@ -842,8 +854,30 @@ def launch_job(self, query, name=None, output_file=None,
         -------
         A Job object
         """
+        compressed_extension = ".gz"
+        format_with_results_compressed = ['votable', 'fits']
+        output_file_with_extension = output_file
+
+        if output_file is not None:
+            if output_format in format_with_results_compressed:
+                # In this case we will have to take also into account the .fits format
+                if not output_file.endswith(compressed_extension):
+                    warnings.warn('WARNING!!! By default, results in "votable" and "fits" format are returned in '
+                                  f'compressed format therefore your file {output_file} '
+                                  f'will be renamed to {output_file}.gz')
```

Comment:
It is from the TAP+ protocol. By default certain type of output formats are returned in compress format

Reply:
If it's a feature of the TAP+ protocol then it would be better to open a separate pull request for implementing this in the `TapPlus` class.

Reply:
Sorry, I'm not sure if I understand here what do you want me to do. TAP+ returning a compressed format is not something that has been implemented now. 
The problem here was that when a user gives a name, for example "myVotable.vot" for the file and he/she does not expect that the file is going to be provided in compressed format and then he/she tries to open the file, the user is going to get and error because the file is compressed. In order to avoid that, we were requested to rename the file with the correct extension if the user didn't provided that at first 'myVotable.vot.gz', so they don't get that error when opening the file.

Reply:
If the `votable` and `fits` files are compressed only by the Gaia archive then this code belongs here. But from your previous answer I got the impression that those file types being compressed is required by the TAP+ standard, and if that is the case then this code needs to go to the `TapPlus` class so that all `astroquery` modules connecting to TAP+ services could benefit from it.

Reply:
It required by TAP+. I can move it to TapPlus for the next implementation but we have dealing with this pull request since past June and I/we would really appreciate if we can move on with this. Specially if as you have suggested, we need to do a different pull request for the TAP+ modifications.  I will take this also into account from now on.

Reply:
This pull request would implement many improvements to `astroquery.gaia`, but at least a few of them seem to be independent from the others. The problem with such a large pull request is that it can only be merged all at once, so if any of the proposed improvements requires some more work then merging all the other improvements has to be delayed too. Opening a separate pull request for each independent improvement would preferable.

Reply:
Dear @eerovaher , @mhsarmiento ,

I think there is a misunderstanding here. Considering previous comments, what you are proposing is that we include that warning as a check in the launch_job method from TAPPlus class, so the rest of our modules could benefit from it. Is that correct? And that this improvement should be included in a different pull request, am I wrong?

If that is the case, I propose this: 

- On Monday, I will extract this piece of code.
- I will create a new branch from astroquery main to include only this feature.
- After testing it and checking that it works for all ESA modules, I will create the pull request. As this is something simple (just a few lines), it should be merged fast.
- Once it is accepted and in astroquery main branch, we will rebase this PR branch to include it and remove the code from Gaia module.

What do you think?

Have a nice weekend!

Reply:
My understanding based on the answers above is that changing file name extensions of compressed formats would be useful for all TAP+ services, and in that case it should be implemented in the `TapPlus` class to avoid code repetition. But I am not familiar with the TAP+ standard so it is possible I have misunderstood something. If I am mistaken and this code is useful only for the Gaia archive then it should stay where it is.

If the code in question does belong in `TapPlus` then I would say it would have been better to open a separate pull request. But the code is already included here, so opening another pull request at this point might not be worth the effort. It is best to ask @keflavich, @bsipocz or @ceb8 what they think.

> As this is something simple (just a few lines), it should be merged fast.

I do not have write access to this repository, so I cannot make any promises about when this or any other pull request gets merged, but I would expect that smaller and simpler pull requests get merged faster.